// lumaai/resources/generations/generations.js
"use strict";
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Generations = void 0;
const resource_1 = require("../../resource.js");
const core_1 = require("../../core.js");
const CameraMotionAPI = __importStar(require("./camera-motion.js"));
const camera_motion_1 = require("./camera-motion.js");
const ImageAPI = __importStar(require("./image.js"));
const image_1 = require("./image.js");
const VideoAPI = __importStar(require("./video.js"));
const video_1 = require("./video.js");
class Generations extends resource_1.APIResource {
    constructor() {
        super(...arguments);
        this.cameraMotion = new CameraMotionAPI.CameraMotion(this._client);
        this.image = new ImageAPI.Image(this._client);
        this.video = new VideoAPI.Video(this._client);
    }
    /**
     * Initiate a new generation with the provided prompt
     */
    create(body, options) {
        return this._client.post('/generations', { body, ...options });
    }
    list(query = {}, options) {
        if ((0, core_1.isRequestOptions)(query)) {
            return this.list({}, query);
        }
        return this._client.get('/generations', { query, ...options });
    }
    /**
     * Remove a specific generation by its ID
     */
    delete(id, options) {
        return this._client.delete(`/generations/${id}`, {
            ...options,
            headers: { Accept: '*/*', ...options?.headers },
        });
    }
    /**
     * Retrieve details of a specific generation by its ID
     */
    get(id, options) {
        return this._client.get(`/generations/${id}`, options);
    }
}
exports.Generations = Generations;
Generations.CameraMotion = camera_motion_1.CameraMotion;
Generations.Image = image_1.Image;
Generations.Video = video_1.Video;
