// lumaai/index.js
"use strict";
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var _a;
Object.defineProperty(exports, "__esModule", { value: true });
exports.UnprocessableEntityError = exports.PermissionDeniedError = exports.InternalServerError = exports.AuthenticationError = exports.BadRequestError = exports.RateLimitError = exports.ConflictError = exports.NotFoundError = exports.APIUserAbortError = exports.APIConnectionTimeoutError = exports.APIConnectionError = exports.APIError = exports.LumaAIError = exports.fileFromPath = exports.toFile = exports.LumaAI = void 0;
const Core = __importStar(require("./core.js"));
const Errors = __importStar(require("./error.js"));
const Uploads = __importStar(require("./uploads.js"));
const API = __importStar(require("./resources/index.js"));
const credits_1 = require("./resources/credits.js");
const ping_1 = require("./resources/ping.js");
const generations_1 = require("./resources/generations/generations.js");
/**
 * API Client for interfacing with the LumaAI API.
 */
class LumaAI extends Core.APIClient {
    /**
     * API Client for interfacing with the LumaAI API.
     *
     * @param {string | undefined} [opts.authToken=process.env['LUMAAI_API_KEY'] ?? undefined]
     * @param {string} [opts.baseURL=process.env['LUMAAI_BASE_URL'] ?? https://api.lumalabs.ai/dream-machine/v1] - Override the default base URL for the API.
     * @param {number} [opts.timeout=1 minute] - The maximum amount of time (in milliseconds) the client will wait for a response before timing out.
     * @param {number} [opts.httpAgent] - An HTTP agent used to manage HTTP(s) connections.
     * @param {Core.Fetch} [opts.fetch] - Specify a custom `fetch` function implementation.
     * @param {number} [opts.maxRetries=2] - The maximum number of times the client will retry a request.
     * @param {Core.Headers} opts.defaultHeaders - Default headers to include with every request to the API.
     * @param {Core.DefaultQuery} opts.defaultQuery - Default query parameters to include with every request to the API.
     */
    constructor({ baseURL = Core.readEnv('LUMAAI_BASE_URL'), authToken = Core.readEnv('LUMAAI_API_KEY'), ...opts } = {}) {
        if (authToken === undefined) {
            throw new Errors.LumaAIError("The LUMAAI_API_KEY environment variable is missing or empty; either provide it, or instantiate the LumaAI client with an authToken option, like new LumaAI({ authToken: 'My Auth Token' }).");
        }
        const options = {
            authToken,
            ...opts,
            baseURL: baseURL || `https://api.lumalabs.ai/dream-machine/v1`,
        };
        super({
            baseURL: options.baseURL,
            timeout: options.timeout ?? 60000 /* 1 minute */,
            httpAgent: options.httpAgent,
            maxRetries: options.maxRetries,
            fetch: options.fetch,
        });
        this.generations = new API.Generations(this);
        this.ping = new API.Ping(this);
        this.credits = new API.Credits(this);
        this._options = options;
        this.authToken = authToken;
    }
    defaultQuery() {
        return this._options.defaultQuery;
    }
    defaultHeaders(opts) {
        return {
            ...super.defaultHeaders(opts),
            ...this._options.defaultHeaders,
        };
    }
    authHeaders(opts) {
        return { Authorization: `Bearer ${this.authToken}` };
    }
}
exports.LumaAI = LumaAI;
_a = LumaAI;
LumaAI.LumaAI = _a;
LumaAI.DEFAULT_TIMEOUT = 60000; // 1 minute
LumaAI.LumaAIError = Errors.LumaAIError;
LumaAI.APIError = Errors.APIError;
LumaAI.APIConnectionError = Errors.APIConnectionError;
LumaAI.APIConnectionTimeoutError = Errors.APIConnectionTimeoutError;
LumaAI.APIUserAbortError = Errors.APIUserAbortError;
LumaAI.NotFoundError = Errors.NotFoundError;
LumaAI.ConflictError = Errors.ConflictError;
LumaAI.RateLimitError = Errors.RateLimitError;
LumaAI.BadRequestError = Errors.BadRequestError;
LumaAI.AuthenticationError = Errors.AuthenticationError;
LumaAI.InternalServerError = Errors.InternalServerError;
LumaAI.PermissionDeniedError = Errors.PermissionDeniedError;
LumaAI.UnprocessableEntityError = Errors.UnprocessableEntityError;
LumaAI.toFile = Uploads.toFile;
LumaAI.fileFromPath = Uploads.fileFromPath;
LumaAI.Generations = generations_1.Generations;
LumaAI.Ping = ping_1.Ping;
LumaAI.Credits = credits_1.Credits;
var uploads_1 = require("./uploads.js");
Object.defineProperty(exports, "toFile", { enumerable: true, get: function () { return uploads_1.toFile; } });
Object.defineProperty(exports, "fileFromPath", { enumerable: true, get: function () { return uploads_1.fileFromPath; } });
var error_1 = require("./error.js");
Object.defineProperty(exports, "LumaAIError", { enumerable: true, get: function () { return error_1.LumaAIError; } });
Object.defineProperty(exports, "APIError", { enumerable: true, get: function () { return error_1.APIError; } });
Object.defineProperty(exports, "APIConnectionError", { enumerable: true, get: function () { return error_1.APIConnectionError; } });
Object.defineProperty(exports, "APIConnectionTimeoutError", { enumerable: true, get: function () { return error_1.APIConnectionTimeoutError; } });
Object.defineProperty(exports, "APIUserAbortError", { enumerable: true, get: function () { return error_1.APIUserAbortError; } });
Object.defineProperty(exports, "NotFoundError", { enumerable: true, get: function () { return error_1.NotFoundError; } });
Object.defineProperty(exports, "ConflictError", { enumerable: true, get: function () { return error_1.ConflictError; } });
Object.defineProperty(exports, "RateLimitError", { enumerable: true, get: function () { return error_1.RateLimitError; } });
Object.defineProperty(exports, "BadRequestError", { enumerable: true, get: function () { return error_1.BadRequestError; } });
Object.defineProperty(exports, "AuthenticationError", { enumerable: true, get: function () { return error_1.AuthenticationError; } });
Object.defineProperty(exports, "InternalServerError", { enumerable: true, get: function () { return error_1.InternalServerError; } });
Object.defineProperty(exports, "PermissionDeniedError", { enumerable: true, get: function () { return error_1.PermissionDeniedError; } });
Object.defineProperty(exports, "UnprocessableEntityError", { enumerable: true, get: function () { return error_1.UnprocessableEntityError; } });
exports = module.exports = LumaAI;
exports.default = LumaAI;
